### DISCUSSION TITLE
Purpose and founding related questions

### DISCUSSION THREAD

**Discussion by jraoult** [mantinedev/mantine#43] [Feedback]
Hey folks,

I've been a long time user of Material UI and I have to say the work and attention to details put in Meantime project is quite impressive. Which brings me to my questions:
- Why did you build this library? Is there any specific goal you were trying to achieve? What's the mindset? Did you extract this from an app in production?
- What's the funding structure? Like I said, I understand what it takes to produce such an impressive amount of work but I didn't really manage to understand how you folks can afford spending so much time on it and still make a living 😉. I guess what I'm trying to gauge is how long this library will be maintained and if you thought about ways to make it last.

Good luck 👍

**Comment by rtivital** [comment]
> Why did you build this library?

I've been building large react applications for the past 4 years and noticed that I'm redoing the same work for each project. I've checked existing solutions and these wasn't any that would fit any of my applications, for example, Material UI does not fit with design, Chakra UI although looking much better provides too complicated components structure and this results in bloated jsx. So I've decided to build my own library that fits all my requirements.

> Is there any specific goal you were trying to achieve?

Well, to build best UI library out there, which I can use in my projects (personal and at work), currently I'm working on ~ 5 new projects each year and all of them fit with Mantine design.

> Did you extract this from an app in production?

Yes and no, several of my projects used the same design and api, but codebase is completely different, all was written from scratch.
I cannot show anything though, everything is under authentication or firewall.

> I understand what it takes to produce such an impressive amount of work but I didn't really manage to understand how you folks can afford spending so much time on it and still make a living

This is what you can achieve if you code during endless zoom meetings 😄. 

> I guess what I'm trying to gauge is how long this library will be maintained and if you thought about ways to make it last.

I'm using mantine in 2 of my large projects at work already, it gives me a lot of interest in supporting it for at least next 3 years. Also I'll try to promote mantine and find contributors and maintainers.

> What's the funding structure?

Right now I do not need any funding – money spoils the idea for me – it's hard to decide how to distribute among contributors and maintainers, how to pay taxes, etc. The main value for me is to connect to community and build a great project which I can be proud of.

**Reply by dimbslmh** [reply]
> I'm using mantine in 2 of my large projects at work already, it gives me a lot of interest in supporting it for at least next 3 years. **Also I'll try to promote mantine and find contributors and maintainers.**

@rtivital Any updates regarding this?

**Reply by rtivital** [reply]
What updates do you expect?

**Reply by dimbslmh** [reply]
Seeing this question was about 9 months ago, I was curious how much has changed since.. 

1 May 2021:
version 0.8.0
Total contributors: 1

7 May 2021:
version 1.0.3
Total contributors: 4

27 Jan 2022:
version 3.6.3
Total contributors: 70

**Reply by rtivital** [reply]
Well, it seems like fine progress for me, nothing really changed much, all my answers are still accurate
